shm/src/map.cpp

```diff
@@ -1 +1,10 @@
 #include "map.hpp"
+
+#include <algorithm>
+
+Island* Map::getIsland(const Coordinates& coordinate) {
+    return &*(std::find_if(islands_.begin(), islands_.end(),
+                           [&](const auto& el) {
+                               return coordinate == el.getPosition();
+                           }));
+}
```

Comment:
Shouldn't we return `nullptr` if island is not found?

```suggestion

    auto island = std::find_if(islands_.begin(), islands_.end(),
                           [&](const auto& island) {
                               return coordinate == island.getPosition();
                           }));
    return island != std::end(islands_) ? &*island : nullptr;
}
```
I've changed also name of lambda parameter. I think '`island` is better. What do you think?

Reply:
Thanks, I forgot about this